samples/collect_data/collect_data.cpp

```diff
@@ -0,0 +1,196 @@
+                std::cout << "unix connection fail:\n";
+                std::cout << unix_conn.last_error_str() << "\n";
+                unix_conn.clear();                
+            }
+
+            if (!is_connected(array_connection, arrSize) && !unix_conn.connect(sockpp::unix_address(android_address))) {
+                std::cout << "android unix connection fail:\n";
+                std::cout << unix_conn.last_error_str() << "\n";
+                unix_conn.clear();                
+            }
+#endif
+            if (is_connected(array_connection, arrSize)) {
+                conn = get_connected(array_connection, arrSize);
+            }
+            std::cout << "\n";
+            my_sleep(std::chrono::seconds(1));
+        }
+
+        std::string request = "<boinc_gui_rpc_request><get_state/></boinc_gui_rpc_request>\3";
+        if (isDebug) {
+            std::cout << "before NULL == conn: " << conn << "\n";
+        }
+        if(NULL == conn) {
+#if !_WIN32
+            unix_conn.close();
+#endif
+            tcp_conn.close();
+            continue;
+        }
+        if (isDebug) {
+            std::cout << "before write: \n";
+        }
+        const ssize_t res = conn->write(request);
+
+        if (isDebug) {
+            std::cout << "after write: " << res << "\n";            
+        }
+        if (res == -1) {
+            conn->close();
+#if !_WIN32
+            unix_conn.close();
+#endif
+            tcp_conn.close();
+            continue;
+        }
+        const std::chrono::time_point<std::chrono::system_clock> before_sleep = std::chrono::system_clock::now();
+        const time_t in_time_t = std::chrono::system_clock::to_time_t(before_sleep);
+        std::string time_str(20, '\0');
+        std::strftime(&time_str[0], time_str.size(), "%Y-%m-%d %H:%M:%S", std::localtime(&in_time_t));
+        std::cout << "now: " << time_str << "\n";
+        std::cout << "request bytes: " << request.size() << "\n";
+        std::cout << "write bytes: " << res << "\n";
+
+        if (isDebug) {
+            std::cout << "after write, before read: \n";
+        }
+
+        ssize_t n = buffer_size;
+        stream.str("");
+        while (n == buffer_size) {
+            n = conn->read(buffer, sizeof(buffer));
+            if (isDebug) {
+                std::cout << "result size: " << n << "\n";
+                std::cout << "buffer: " << buffer << "\n";
+            }
+            if (n > 0) {
+                std::string str(buffer, buffer + n);
+                stream << str;
+            }
+        }
+        std::string str = stream.str();
+        str = str.substr(0, str.size() - 1);
+        std::cout << "read bytes: " << str.size() << "\n";
+        if(str.empty()) {
+            continue;
+        }
+        std::ofstream outFile;
+        std::stringstream file_name;
+        file_name.str("");
+        file_name << prefix_file_name << suffix_file_idx << ".txt";
+        outFile.open(file_name.str());
```

Comment:
File will be overwritten every time. Is this by intention?

Reply:
Files will overwritten every run. It intention. the idea, that the app will send it over the server then delete the files. 
This is an example, not a real app.

Reply:
Usually files/results are sent to server when task is done. So it's better to append to the file, otherwise server will just receive the last result that is probably not what you're expecting.

Reply:
ok. I understand it now. I will do it.